src/Features/LanguageServer/ProtocolUnitTests/SignatureHelp/SignatureHelpTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Test.Utilities;
+using Xunit;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.UnitTests.SignatureHelp
+{
+    public class SignatureHelpTests : AbstractLanguageServerProtocolTests
+    {
+        [Fact]
+        public async Task TestGetSignatureHelpAsync()
+        {
+            var markup =
+@"class A
+{
+    void M()
+    {
+        M2({|caret:|}'a');
+    }
+    /// <summary>
+    /// M2 is a method.
+    /// </summary>
+    int M2(string a)
+    {
+        return 1;
+    }
+
+}";
+            var (solution, locations) = CreateTestSolution(markup);
+            var expected = new LSP.SignatureHelp()
+            {
+                ActiveParameter = 0,
+                ActiveSignature = 0,
+                Signatures = new LSP.SignatureInformation[] { CreateSignatureInformation("int A.M2(string a)", "M2 is a method.", "a", "") }
+            };
+
+            var results = await RunGetSignatureHelpAsync(solution, locations["caret"].First());
+            AssertSignatureHelpEquals(expected, results);
```

Comment:
>AssertSignatureHelpEquals [](start = 12, length = 25)

Seems a bit verbose to have to write all the Assert*Equal methods to validate the results. 
Sometimes it is easier to write an inspect method that formats the interesting properties of the objects to strings and then compare strings. 
E.g.: 

[https://github.com/tmat/roslyn/blob/EncManagerUpdateP2/src/EditorFeatures/Test/EditAndContinue/EditSessionActiveStatementsTests.cs#L278-L286](https://github.com/tmat/roslyn/blob/EncManagerUpdateP2/src/EditorFeatures/Test/EditAndContinue/EditSessionActiveStatementsTests.cs#L278-L286)

Reply:
oh that's a good idea.  Since the output should be a JSON object eventually (and the types are already convertible), I can just compare the the result to an expected json string.  will update, thanks!